sql/hive-thriftserver/v1.2/src/main/java/org/apache/hive/service/cli/OperationState.java

```diff
@@ -32,7 +32,8 @@
   CLOSED(TOperationState.CLOSED_STATE, true),
   ERROR(TOperationState.ERROR_STATE, true),
   UNKNOWN(TOperationState.UKNOWN_STATE, false),
-  PENDING(TOperationState.PENDING_STATE, false);
+  PENDING(TOperationState.PENDING_STATE, false),
+  TIMEDOUT(TOperationState.CANCELED_STATE, true); //do not want to change TOperationState in hive 1.2
```

Comment:
What does `//do not want to change TOperationState in hive 1.2` means?

Reply:
The class of TOperationState was generated by thift, so we should not change it directly

Reply:
Yes. Only `hive-2.3` support timeout.